ASL demo internal PR

mpashchenkov opened the pull request.

Commit:
Merge pull request #1924 from openvinotoolkit/release

Merge OpenVINO toolkit 2021.2 content into master

Commit:
Merge pull request #2238 from openvinotoolkit/release

Merge OpenVINO toolkit 2021.3 content into master

Wovchena committed:
Merge pull request #2557 from openvinotoolkit/release

Merge OpenVINO toolkit 2021.4 content into master

Wovchena committed:
ci: bump OpenVINO version

Wovchena committed:
Merge pull request #2570 from Wovchena/ci-bump-OpenVINO-version

ci: bump OpenVINO version

Commit:
Gesture recognition demo

eizamaliev committed:
Add monodepth_demo

mpashchenkov commented:
@dmatveev

@adanpin, if interest is exist.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/CMakeLists.txt:
```diff
@@ -0,0 +1,12 @@
+# Copyright (C) 2018-2019 Intel Corporation
```
`-2019`?

mpashchenkov replied:
Fixed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/exponential_averager.hpp:
```diff
@@ -0,0 +1,16 @@
+private:
+    double smoothingFactor;
+    double value;
+};
```
Is it taken from the original demo and is this the way it is called in the original demo?

mpashchenkov replied:
I get this from Gaze estimation. This will be removed, because OMZ get unified FPS counter soon.

mpashchenkov replied:
Unified metrics have appeared. Will be removed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,52 @@
+    }
+};
+
+} // namespace custom
```
I believe every sample have it now, is it worth putting it to some common place?

mpashchenkov replied:
All demos use this stream_source (IFD soon). It is good point put this to common cpp_gapi folder.

mpashchenkov replied:
https://github.com/openvinotoolkit/open_model_zoo/pull/2778

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/tracker.hpp:
```diff
@@ -0,0 +1,300 @@
+#include <vector>
+#include <utils/kuhn_munkres.hpp>
+
+struct TrackedObject {
+    cv::Rect rect;
+    float confidence;
+
+    int object_id;
+
+    size_t frame_idx;
```
Can this structure be just plain? E.g. default field initializers and no constructor?

mpashchenkov replied:
Fixed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,229 @@
+
+            /** Controls **/
+            int key = cv::waitKey(1);
+            if      (key == 27) break;  //esc
+            if      (key == 48) id = 0; // person id
+            else if (key == 49) id = 1;
+            else if (key == 50) id = 2;
+            else if (key == 51) id = 3;
+            else if (key == 52) id = 4;
+            else if (key == 53) id = 5;
+            else if (key == 54) id = 6;
+            else if (key == 55) id = 7;
+            else if (key == 56) id = 8;
+            else if (key == 57) id = 9;
+            else if (key == 13) out_label_number = -1; // reset last gesture
```
magik numbers here?

mpashchenkov replied:
Key codes for 0...9 buttons

mpashchenkov replied:
Added comments.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,229 @@
+        cv::GArray<TrackedObject> tracked = custom::PersonTrack::on(in, objects);
+
+        /** Create batch for ROIs **/
+        cv::GArray<cv::GMat> batch = custom::ActionRecognisePreprocessing::on(in, tracked, id_flag, ar_net_shape);
```
Recogni**z**e

mpashchenkov replied:
I'll update all kernels names.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+    const int src_roi_center_x = 0.5 * (person_roi.tl().x + person_roi.br().x);
+    const int src_roi_center_y = 0.5 * (person_roi.tl().y + person_roi.br().y);
+
+    float height_scale = float(in_size.height) / float(scale);
+    float width_scale = float(in_size.width) / float(scale);
```
also `const`?

mpashchenkov replied:
Fixed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+    CV_Assert(height_scale < 1.0f);
+    CV_Assert(width_scale < 1.0f);
+
+    int min_roi_size = std::min(roi_height, roi_width);
+    int trg_roi_height = height_scale * min_roi_size;
+    int trg_roi_width = width_scale * min_roi_size;
```
also `const`?

mpashchenkov replied:
Fixed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+    float height_scale = float(in_size.height) / float(scale);
+    float width_scale = float(in_size.width) / float(scale);
+
+    CV_Assert(height_scale < 1.0f);
+    CV_Assert(width_scale < 1.0f);
```
Is this method called in some hot inner loop? Is it performance critical? Should it be `DbgAssert` instead?

mpashchenkov replied:
Changed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+                                        conf);
+            }
+    }
+};
```
Can this be built atop of our parseSSD?

mpashchenkov replied:
Unfortunately this "parseSSD"  has OBJECT_SIZE = 5 (not 7).

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+        label_number = -1;
+        if (!asl_result.empty()) {
+            const float* data = asl_result[0].ptr<float>();
+            // Find more suitabl action
```
Suitable

mpashchenkov replied:
Fixed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+            const float* data = asl_result[0].ptr<float>();
+            // Find more suitabl action
+            float max_weight = 0.f;
+            const auto num_classes = asl_result[0].size().width;
```
Why we only access 0th element? What's the idea behind this operation?

mpashchenkov replied:
I think, Daniil's idea is avoid batch size in outMeta and use instead cv::empty_array_desc. Can it be changed to GOpaque?
0th element - the only one Mat for all cases of demo.

mpashchenkov replied:
Changed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+            // Find more suitabl action
+            float max_weight = 0.f;
+            const auto num_classes = asl_result[0].size().width;
+            for(int i = 0; i < num_classes; ++i) {
+                if (data[i] > max_weight && data[i] > ar_threshold) {
+                    max_weight = data[i];
+                    label_number = i;
+                }
+            }
```
What if nothing is found? Is `-1` a correct result then?

dmatveev replied:
Also, I believe this loop is just `minMaxLoc` - you just find the maximum value in the array and then compare it against `ar_threshold` only once.

mpashchenkov replied:
Changed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+struct ARStateParams {
+    std::list<cv::Mat> stored_frames;
+    int last_id;
+};
```
Please put this structure closer to the kernel where it is used.

dmatveev replied:
Also - may it be defined right in the `_ST` kernel? And referred to as `OCVPreprocessing::ARStateParams ` ?

mpashchenkov replied:
?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+                    const cv::Scalar& net_size,
+                          std::vector<cv::Mat> &prepared_mat,
+                    ARStateParams& ar_state) {
+        const int channels = int(net_size[0]);
+        const int batch = int(net_size[1]);
+        const int height = int(net_size[2]);
+        const int width = int(net_size[3]);
```
y `int{}` ? if a cast is implied here, I'd recommend to indicate it with `_cast<>`

mpashchenkov replied:
But it is equal `static_cast<int>`, isn't it? It isn't C-style `(int)number`.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+        const int height = int(net_size[2]);
+        const int width = int(net_size[3]);
+        cv::Size sz_dst(height, width);
+        if (ar_state.stored_frames.size() < size_t(batch)) {
```
is this some OMZ's conversion to make casts via `T()` ?

mpashchenkov replied:
Removed

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+        const int width = int(net_size[3]);
+        cv::Size sz_dst(height, width);
+        if (ar_state.stored_frames.size() < size_t(batch)) {
+            ar_state.stored_frames.emplace_back(in_frame.clone());
```
ah so you cache only input frames as a whole - and then do preprocessing for those again and again?

mpashchenkov replied:
Preprocessing is required for whole batch

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+        if (ar_state.stored_frames.size() < size_t(batch)) {
+            ar_state.stored_frames.emplace_back(in_frame.clone());
+        } else {
+            if(prepared_mat.size() == 0) {
```
`if (`

mpashchenkov replied:
Fixed.

mpashchenkov replied:
Removed

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+            if(prepared_mat.size() == 0) {
+                cv::Mat mat_to_process;
+                mat_to_process.create({1, channels, batch, height, width}, CV_32F);
+                // FIXME: Why is GArray result?
```
Probably it was done to avoid the automatic preprocessing which kicks in here.

mpashchenkov replied:
Yes. It is.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+        } else {
+            if(prepared_mat.size() == 0) {
+                cv::Mat mat_to_process;
+                mat_to_process.create({1, channels, batch, height, width}, CV_32F);
```
Why always allocate a new one? This may be expensive, I'd suggest putting it to the state.

mpashchenkov replied:
It is state parameter now.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+            ar_state.stored_frames.push_back(in_frame.clone());
+
+            if (size_t(*runtime_id) < tracked_persons.size()) {
+                ar_state.last_id = *runtime_id;
```
if `runtime_id` is altered from another thread (and likely it is), this may be not very correct

mpashchenkov replied:
main() changes runtime_id .

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+                ar_state.last_id = *runtime_id;
+            }
+            int step = 0;
+            for(const auto& frame : ar_state.stored_frames) {
```
`for (`

mpashchenkov replied:
Removed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+                cv::Mat crop, cvt, resized;
+                if (tracked_persons.size() > 0 &&
+                    size_t(ar_state.last_id) < tracked_persons.size()) {
+                    cv::Rect roi = tracked_persons.at(ar_state.last_id).rect;
```
so may this ROI change for an old frame?

dmatveev replied:
your track is calculated for the current frame, isn't it? Is it correct to use this ROI on older frames?

mpashchenkov replied:
Yes, it is correct. Person have to stay in the frame.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,196 @@
+                cv::Mat matG = different_channels[1];
+                cv::Mat matR = different_channels[2];
+
+                std::copy_n(matR.begin<float>(), height * width,
+                            prepared_mat[0].begin<float>() + 0 * batch * height * width
+                                + step * height * width);
+                std::copy_n(matG.begin<float>(), height * width,
+                            prepared_mat[0].begin<float>() + 1 * batch * height * width
+                                + step * height * width);
+                std::copy_n(matB.begin<float>(), height * width,
+                            prepared_mat[0].begin<float>() + 2 * batch * height * width
+                                + step * height * width);
```
What I'd expect here is
1. The preprocessing part (`crop`, `resize`, `covertTo`, `split`) - happening only once per a new frame - if it is possble.
2. **Only** copy_n happenning for all (past+new) preprocessed frames to form the resulting blob.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/exponential_averager.cpp:
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "exponential_averager.hpp"
+
+ExponentialAverager::ExponentialAverager(double smoothingFactor, double initValue):
+                     smoothingFactor(smoothingFactor), value(initValue) {
+}
+
+double ExponentialAverager::getAveragedValue() const {
+    return value;
+}
+
+void ExponentialAverager::updateValue(double newValue) {
+    value = smoothingFactor * newValue + (1. - smoothingFactor) * value;
+}
```
This thing looks quite tiny, it is used everywhere or just in a single place? If the latter, I'd drop a separate file here and put it right to the place where it is used.

mpashchenkov replied:
Will be removed.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/utils.cpp:
```diff
@@ -0,0 +1,64 @@
+        throw std::logic_error("Gesture file doesn't open.");
+    }
+    return labels;
+}
```
no newline here

mpashchenkov replied:
Added

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/src/visualizer.cpp:
```diff
@@ -0,0 +1,101 @@
+                cv::destroyWindow(storage_window_name_);
+            }
+        }
+    }
```
no newline here either, also the alignment "hangs"

mpashchenkov replied:
Added

dmatveev commented on demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp:
```diff
@@ -327,6 +327,7 @@ std::shared_ptr<FaceRecognizer> processingFaceGallery(const cv::gapi::GNetPackag
         int id = 0;
         for (const auto& item : fn) {
             std::string label = item.name();
+            std::cout << label << std::endl;
```
is it required? Seems like an unstructured random output

mpashchenkov replied:
No. Removed.

Commit:
IFD refactoring + common stream source

eizamaliev committed:
Fix

eizamaliev committed:
Update tests

eizamaliev committed:
Add suggestions

eizamaliev committed:
Apply suggestions from code review

Co-authored-by: Anzhella Pankratova <anzhella.pankratova@intel.com>

Wovchena committed:
Merge pull request #2749 from openvinotoolkit/release

Merge OpenVINO toolkit 2021.4.1 content into master

DmitryGrayscale committed:
Use strict major version v0.24 for scikit-learn due to py3.6 support drop in v1.0 (#2744)

eaidova committed:
AC: port important bugfixes to release (#2763)

* AC: port important bugfixes to release

* set metric profiler for coco_orig_precision

anzhella-pankratova committed:
Update demos/common/python/models/monodepth.py

Commit:
update documentation for retinanet-tf (#2769) (#2771)

anzhella-pankratova committed:
Merge remote-tracking branch 'b/develop' into HEAD

anzhella-pankratova committed:
Move monodepth.py to openvino/model_zoo/model_api/

anzhella-pankratova committed:
Update tests for pedestrian_tracker_demo

anzhella-pankratova committed:
Update demos/tests/cases.py

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

anzhella-pankratova committed:
Fix: str -> ModelArg

Wovchena committed:
Merge pull request #2788 from anzhella-pankratova/fix_tests_pedestrian_tracker

Update tests for pedestrian_tracker_demo

Commit:
Fixed pylint(2.10.2) errors in AC files (#2789)

* Fixed pylint(2.10.2) errors in AC files

* Fix comment

Commit:
Raw comparison for reference (#2785)

Commit:
Model Quantizer: update pot import (#2790)

* update pot import

* preserve old import

eaidova committed:
AC: port some missed bugfixes to new layout (#2791)

Wovchena committed:
Push public models

Wovchena committed:
Merge pull request #2742 from eizamaliev/monodepth_model_api

[ModelAPI] monodepth_demo

Wovchena committed:
demos/tests: resurect missing tests

eaidova committed:
AC: fix action detection adapter (#2793)

Wovchena committed:
ci: change parrent

Wovchena committed:
Merge pull request #2796 from Wovchena/cichangeparrent

ci: change parent

Wovchena committed:
Merge pull request #2794 from Wovchena/demostestsresurectmissingtests

demos/tests: resurrect missing tests

eaidova committed:
AC: solve device selection if several devices available (#2795)

anzhella-pankratova committed:
Replace deprecated warn -> warning

anzhella-pankratova committed:
gtp2_text_prediction_demo documentation fix

anzhella-pankratova committed:
Fix test cases for face_recognition_demo

anzhella-pankratova committed:
Delete unused labels files

Commit:
update paths to mo and pot

eaidova committed:
AC: support sentence similarity benchmarking (#2797)

* AC: support sentense similarity task

* add metric and subset logic

* add docs, align distance scores with sklearn

anzhella-pankratova committed:
Move set_inputs_preprocessing() to ImageModel

eaidova committed:
AC: support reading npz data for speech regression pipeline (#2798)

eaidova committed:
AC: update convert for STS-B (#2800)

anzhella-pankratova committed:
Update path to the labels in whiteboard_inpainting_demo

anzhella-pankratova committed:
Inherit CTPN and YOLOX from DetectionModel

anzhella-pankratova committed:
Update Deblurring class

Wovchena committed:
Merge pull request #2801 from anzhella-pankratova/python_small_fixes

Python demos: minor fixes

anzhella-pankratova committed:
[Tests] Set precisions for model testing via --precisions parameter (#2779)

* Add --precisions parameter to run_tests

* Set default value for self.precisions as []

* Update set_precisions: testing models with precisions for which model has IR

* Fixes & suggestions

* Change default precision

* Remove add_case

* set_precisions() refactoring

* Fixes

* Make update_case() as Demo method

* Modify update_case()

vladimir-dudnik committed:
Added mxnet==1.2.0 to requirements-conversion.txt for Demos Windows 10 Job (#2701) (#2802)

* Removed win32 exception for mxnet

* Added mxnet==1.2.0 for win32

Co-authored-by: Nikita Ignatovsky <nikita.ignatovsky@intel.com>

Commit:
lload mo and pot from entrypoints

Wovchena committed:
Added mxnet==1.2.0 to requirements-conversion.txt for Demos Windows 10 Job (#2701) (#2802) (#2803)

* Removed win32 exception for mxnet

* Added mxnet==1.2.0 for win32

Co-authored-by: Nikita Ignatovsky <nikita.ignatovsky@intel.com>

Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>
Co-authored-by: Nikita Ignatovsky <nikita.ignatovsky@intel.com>

eaidova committed:
model docs: fix info about output format for bin models (#2806)

Commit:
G-API. Face detection MTCNN demo. (#2657)

* Apply suggestions from code review

commit suggestions

Co-authored-by: Eduard Zamaliev <eduard.zamaliev@intel.com>

* G-API version aligned with OMZ

* Added last line to README

* add model_keys

Co-authored-by: Eduard Zamaliev <eduard.zamaliev@intel.com>
Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

eaidova committed:
AC: remove allmentions about FPGA (#2807)

Commit:
update README, use shutil

Wovchena committed:
Merge pull request #2804 from jkamelin/jk/new_irc_structure

tools/model_tools: Update mo and pot calls

eaidova committed:
AC: skip G-API launcher tests if opencv without OV (#2808)

Wovchena committed:
Merge pull request #2792 from Wovchena/pushasubestofpublicmodels

Push a subset of public models

ygnn123 committed:
Added 98 landmarks detection net. Added script to accuracy checker fo… (#2457)

* Added 98 landmarks detection net. Added script to accuracy checker for validation

* Update models/intel/face-landmarks-98-detection-0001/description/face-98-landmarks-detection-0001.md

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* Update tools/accuracy_checker/accuracy_checker/representation/face_98_landmarks_representation.py

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* Update tools/accuracy_checker/accuracy_checker/representation/face_98_landmarks_representation.py

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* Update models/intel/face-landmarks-98-detection-0001/description/face-98-landmarks-detection-0001.md

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* fiw trailing whitespces

* fix ac config

* to json

* Update tools/accuracy_checker/accuracy_checker/annotation_converters/coco_facial_landmarks.py

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* fix comments

* fix comments

* fix comments

* fix pylint

* add model to index.md

* fix pylint again

* Update models/intel/face-landmarks-98-detection-0001/README.md

Co-authored-by: Anna Mironova <anna.mironova@intel.com>

* Update models/intel/face-landmarks-98-detection-0001/README.md

Co-authored-by: Anna Mironova <anna.mironova@intel.com>

* Update README.md

* Update models/intel/face-landmarks-98-detection-0001/README.md

Co-authored-by: Anna Mironova <anna.mironova@intel.com>

* Update models/intel/face-landmarks-98-detection-0001/README.md

Co-authored-by: Anna Mironova <anna.mironova@intel.com>

* Fix metrics

* add descriptions

* Update tools/accuracy_checker/accuracy_checker/annotation_converters/README.md

Co-authored-by: Anna Mironova <anna.mironova@intel.com>

* Update models/intel/face-landmarks-98-detection-0001/README.md

Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

* Update models/intel/face-landmarks-98-detection-0001/README.md

Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

* Replace imgs. Fix device support file

* add lost image

* fix pr comments

* rem empty space

* Update tools/accuracy_checker/accuracy_checker/annotation_converters/README.md

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* fix naming

* Update models/intel/face-landmarks-98-detection-0001/accuracy-check.yml

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* Update models/intel/device_support.md

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* fix import

* fix

* fix pylint

* fix pylint

* fix readme

* fix naming

* add dataset info

* global commit

* fixing

* fixing

* fixing

* fixing

* fixing

* change image

* rebase to develop

* Update coco_facial_landmarks.py

Co-authored-by: yurygoru <yury.gorunov@intel.com>
Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>
Co-authored-by: Anna Mironova <anna.mironova@intel.com>
Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

Wovchena committed:
Merge remote-tracking branch 'upstream/release' into merge-release-into-develop

Commit:
AC support for jpeg artifacts removal models (#2812)

* ac support for jpeg artifacts removal models

* remove unused libs

* update copyright and names for compression preprocessing

* update init for preprocessing

vladimir-dudnik committed:
extend test cases for VPU (#2810)

Commit:
update sha384

mpashchenkov commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+                 *  First image is 0, counter is 1 - 16(11) **/
+                ptr[(batch_size + 1) * frame.total() + 0] = cv::Vec3b(0, counter, 0);
+            } else if(counter == batch_size) { // batch is filled already rewrite memory of batch
+                /** Adding of new image to batch. Now it is placed on ferst element place. **/
```
first

Wovchena committed:
Merge pull request #2811 from Wovchena/merge-release-into-develop

Merge release into develop

mpashchenkov commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,211 @@
+            int minIdx = 0, maxIdx = 0;
+            const float* data = asl_result[0].ptr<float>();
+            // Find more suitable action
+            cv::minMaxIdx(asl_result[0].reshape(1, {int(asl_result.total())}), &min, &max, &minIdx, &maxIdx);
```
asl_result.total() --> asl_result[0].total()

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+    int batch_fps = 0;  // constant FPS for batch
+    cv::Mat batch;
+    int first_el = 0;   // number of first image of batch
+    int counter = 0;    // amount of images in batch
```
how is about change name  from `counter` to `images_in_batch_count` or some thing like to keep code self-documented

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+            return false;
+        }
+        /** Copy fast/real/current frame from VideoCapture to batch memory as 17th image **/
+        std::copy_n(frame.begin<cv::Vec3b>(),
```
suggest to introduce function for incapsulate that magic with descriptive name and for other like `ptr[(batch_size + 1) * frame.total()] = ...`  magic

sivanov-work replied:
something like `pack_image_in_network_input`

sivanov-work replied:
or may be introduce new data structure which incapsulate operation  with this `batch`
in initial look it seems as some kind of `circular buffer`

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+                std::copy_n(frame.begin<cv::Vec3b>(), frame.total(),
+                            batch.begin<cv::Vec3b>() + first_el * frame.total());
+                /** And first image is next image (second image in the past)**/
+                if (++first_el > (batch_size - 1)) {
```
popular trick here
`first_el = (first_el +1) % batch_size`

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+        batch.create(cv::Size{frame.cols, (batch_size + 1 + 1) * frame.rows}, CV_8UC3);
+
+        /** Copy fast/real/current frame from VideoCapture to batch memory as 17th image **/
+        std::copy_n(frame.begin<cv::Vec3b>(), frame.total(),
+                    batch.begin<cv::Vec3b>() + batch_size * frame.total());
```
Is `batch_size == 0` or `1` possible? 
suggest to add throwing_exception in Source constructor

dmatveev replied:
Why copy via `Vec3b`?? It looks fairly inefficient. Suggest to use pointers instead (or even `cv:::Mat::copyTo(rowrange)`).

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+         *  sacrifice allows avoid extra copy_n in the future **/
+        batch.create(cv::Size{frame.cols, (batch_size + 1 + 1) * frame.rows}, CV_8UC3);
+
+        /** Copy fast/real/current frame from VideoCapture to batch memory as 17th image **/
```
what is `17th` ?

dmatveev replied:
> fast/real/current

Are these three different ones or it is the same frame? To avoid confusion, I'd call it with a single name throughout the code

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+
+            /** Controls **/
+            int key = cv::waitKey(1);
+            if      (key == 27) break;  // (esc button) exit
```
i do not know why, but as for me HEX digits looks more friendly
(please skip this comment)

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/tracker.cpp:
```diff
@@ -0,0 +1,359 @@
+#include <set>
+
+cv::Point Center(const cv::Rect &rect) {
+    return cv::Point(static_cast<int>(rect.x + rect.width * 0.5),
```
i remember error in MTCNN implementation where negative value cause integer overflow...
Can we put some ASSERT check here

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/tracker.cpp:
```diff
@@ -0,0 +1,359 @@
+
+TrackerParams::TrackerParams()
+    : min_track_duration(25),
+      forget_delay(150),
```
sec, msec?

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/tracker.cpp:
```diff
@@ -0,0 +1,359 @@
+      drop_forgotten_tracks(true),
+      max_num_objects_in_track(300),
+      averaging_window_size_for_rects(1),
+      averaging_window_size_for_labels(1) {}
```
how this values affect quality or some other stuff? Do we need to calibrate them later?
How's about to put some comment here ?

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/tracker.cpp:
```diff
@@ -0,0 +1,359 @@
+
+void Tracker::SolveAssignmentProblem(
+        const std::set<size_t> &track_ids, const TrackedObjects &detections,
+        std::set<size_t> *unmatched_tracks, std::set<size_t> *unmatched_detections,
```
may be references instead pointer would be better?

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/tracker.hpp:
```diff
@@ -0,0 +1,292 @@
+    /// \param[in] timestamp Timestamp must be positive and measured in
+    /// milliseconds
+    ///
+    void Process(const cv::Mat &frame, const TrackedObjects &detections);
```
mix of Upper/Lower cases between methods

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/tracker.hpp:
```diff
@@ -0,0 +1,292 @@
+    /// \brief Returns recently detected objects.
+    /// \return recently detected objects.
+    ///
+    const TrackedObjects &detections() const;
```
from declaration it is not clear for me what difference with https://github.com/mpashchenkov/open_model_zoo/pull/4/files#diff-926309cf367c96d92a63c8959c910002e57266f02a04feb0de86a420945fff3aR188

and why the latter declaration returns object neither cons reference

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/tracker.hpp:
```diff
@@ -0,0 +1,292 @@
+    /// \brief Get active tracks to draw
+    /// \return Active tracks.
+    ///
+    std::unordered_map<size_t, std::vector<cv::Point>> GetActiveTracks() const;
```
i would suggest

```
using Track = std::vector<cv::Point>
using TrackId = size_t;
```
then
```
std::unordered_map<TrackId, Track> getActiveTrack() ;
```
is becoming more self-documented

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+                                                                                    batch_constant_FPS),
+                                   current_person_id));
+
+        std::string mainWindowName = "Gesture Reognition demo G-API";
```
Re**c**ognition

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,211 @@
+        const int height = int(net_size[2]);
+        const int width = int(net_size[3]);
+        const auto ptr = batch.ptr<cv::Vec3b>();
+        const bool is_filled = ptr[(duration + 1) * image_size.area()][2] == 1 ? true : false;
```
if think `...[2] == 1`;   is `bool` already

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -0,0 +1,211 @@
+            const float* data = asl_result[0].ptr<float>();
+            // Find more suitable action
+            cv::minMaxIdx(asl_result[0].reshape(1, {int(asl_result.total())}), &min, &max, &minIdx, &maxIdx);
+            if (data[maxIdx] > ar_threshold) {
```
`const float* data = asl_result[0].ptr<float>();` after reshape does it retain is `data` pointer safe? ( i mean is there are reallocation inside?)

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+                ptr[(batch_size + 1) * frame.total()] = cv::Vec3b(first_el, counter, 1);
+            }
+        }
+        data = batch.clone();
```
as I understand idea correctly a network requires 15 frame to make suggestion about movement (initial pose... final pose)
but we are returned from `pull` method every time and 90% duration and it returns not filled `batch`  without 15 filled frames in itself. So in which part we waiting for the last 15th frame right before network starting? does it happen in another thread in custom kernel?

sivanov-work replied:
and what happens if source is get stucked and cannot  satisfy 15 fps criteria? do we need to reset batch buffer and fill it it from beginning?

sivanov-work reviewed (commented):
LGTM,
But I suppose that incapsulate working with batch in standalone data_structure/throttler as separate class would be more easier for understanding its logic

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/gesture_recognition_gapi.hpp:
```diff
@@ -0,0 +1,64 @@
+    std::cout << "    -s                       " << samples_dir_message << std::endl;
+    std::cout << "    -t                       " << thresh_output_message << std::endl;
+    std::cout << "    -u                       " << utilization_monitors_message << std::endl;
+}
```
Is it aligned with what's in README?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/custom_kernels.hpp:
```diff
@@ -0,0 +1,71 @@
+    }
+};
+
+G_API_OP(BoundingBoxExtract,
```
I'd name it `ExtractBoundingBox` to follow the natural language, because y not?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/custom_kernels.hpp:
```diff
@@ -0,0 +1,71 @@
+    }
+};
+
+G_API_OP(PersonTrack,
```
Same approach here, -> TrackPerson

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/custom_kernels.hpp:
```diff
@@ -0,0 +1,71 @@
+    }
+};
+
+G_API_OP(ConstructClip,
```
Hah and this is a good one!

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/custom_kernels.hpp:
```diff
@@ -0,0 +1,71 @@
+G_API_OP(ConstructClip,
+         <cv::GArray<cv::GMat>(const cv::GMat,
+                               const cv::GArray<TrackedObject>,
+                               cv::GOpaque<std::shared_ptr<int>>,
```
what's the reason to make it `shared_ptr` -? should this be a pointer to atomic instead?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/custom_kernels.hpp:
```diff
@@ -0,0 +1,71 @@
+    }
+};
+
+G_API_OP(ARPostprocessing,
```
PostprocessAR? Also, what is AR here? (I believe not the AR people usually think of)

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+class CustomCapSource : public cv::gapi::wip::IStreamSource
+{
+public:
+    explicit CustomCapSource(std::shared_ptr<ImagesCapture>& cap, const int batch_size, const int batch_fps)
```
What's the reason to take `cap` by a non-`const` reference here?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+    std::shared_ptr<ImagesCapture> cap;
+    int batch_size = 0; // size of batch
+    int batch_fps = 0;  // constant FPS for batch
+    cv::Mat batch;
```
I believe this `Mat` should be commented too, especially if this `Mat`  is not a plain simple `Mat` but hides some "batch" inside. What is its memory layout then?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+    void prep() {
+        GAPI_Assert(batch.empty());
+        cv::Mat frame = cap->read();
+        if (!frame.data) {
+            GAPI_Assert(false && "Couldn't grab the frame");
+        }
```
What if the input stream is empty? This may be valid situation and the pipeline won't produce anything in this case (the very first `pull()` returns `false`).

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+        ++counter;
+
+        /** Adding of info about batch **/
+        cv::Vec3b* ptr = batch.ptr<cv::Vec3b>();
+        /** Vec3b additional element contains information about place of first frame and
+         *  quantity of frames in batch (asl works with filled batch) **/
```
You can use `metadata` for this instead.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -0,0 +1,120 @@
+        cv::Vec3b* ptr = batch.ptr<cv::Vec3b>();
+        /** Vec3b additional element contains information about place of first frame and
+         *  quantity of frames in batch (asl works with filled batch) **/
+        ptr[(batch_size + 1) * frame.total()] = cv::Vec3b(0, counter, 0); // position of first and counter
```
Is there any explanation what this batch is and how it is formed? I believe this procedure can be further streamlined.

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/tracker.hpp:
```diff
@@ -0,0 +1,292 @@
+#include <utils/kuhn_munkres.hpp>
+
+struct TrackedObject {
+    cv::Rect rect = cv::Rect();
```
Not sure if it is required, a default constructor should be called automatically?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/include/tracker.hpp:
```diff
@@ -0,0 +1,292 @@
+    cv::Size frame_size_;
+
+    size_t frame_idx_ = 0;
+};
```
can this tracker be simply reused from the original demo?

dmatveev replied:
or the original demo is python only?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+}
+
+namespace nets {
+G_API_NET(PersoneDetection, <cv::GMat(cv::GMat)>, "perspne_detection");
```
What is `perspne`?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0,
+            std::numeric_limits<size_t>::max(), stringToSize(FLAGS_res));
+        const auto tmp = cap->read();
+        cap.reset();
```
why call `reset` here?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+        cv::GMat batch;
+        cv::GOpaque<std::shared_ptr<int>> selected_person_id;
+
+        cv::GMat fast_frame = custom::GetFastFrame::on(batch, frame_size);
```
fast or last?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+        cv::GMat fast_frame = custom::GetFastFrame::on(batch, frame_size);
+
+        /** Person detection **/
+        cv::GMat detections = cv::gapi::infer<nets::PersoneDetection>(fast_frame);
```
Likely `Person` not `Persone`

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+        cv::GArray<cv::GMat> clip = custom::ConstructClip::on(batch, tracked, selected_person_id, ar_net_shape, frame_size);
+
+        /** Action recognition **/
+        cv::GArray<cv::GMat> actions = cv::gapi::infer2<nets::ActionRecognition>(batch, clip);
```
If `batch` is not used by this network (and it is not - since no ROIs passed), I'd pass `last_frame` instead

dmatveev replied:
however, maybe passing `batch` is ok too from the logical standpoint (as `clip` is created from the batch).

dmatveev replied:
Also, can `clip` be batch itself to run it via plain `infer` with no lists?

dmatveev commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -0,0 +1,236 @@
+            /** Controls **/
+            int key = cv::waitKey(1);
+            if      (key == 27) break;  // (esc button) exit
+            else if (key == 48) id = 0; // (0 button) person id
+            else if (key == 49) id = 1; // (1 button) person id
+            else if (key == 50) id = 2; // (2 button) person id
+            else if (key == 51) id = 3; // (3 button) person id
+            else if (key == 52) id = 4; // (4 button) person id
+            else if (key == 53) id = 5; // (5 button) person id
+            else if (key == 54) id = 6; // (6 button) person id
+            else if (key == 55) id = 7; // (7 button) person id
+            else if (key == 56) id = 8; // (8 button) person id
+            else if (key == 57) id = 9; // (9 button) person id
```
```
else if (key >= 48 && key <= 57) id = key - 48;
```

vladimir-dudnik committed:
fix for speech noise suppression doc (#2820)

* add section for speech noise suppression model

* capitalize letters

eaidova committed:
AC: support custom asr model (#2822)

eaidova committed:
AC: update sklearn (#2821)

eaidova committed:
AC: support metric refs extraction (#2823)

SDxKeeper committed:
Fix typo (#2825)

eaidova committed:
AC: make padding configurable for squad dataset (#2826)

eaidova committed:
AC: support target backend filtering (#2828)

* AC: support target backend filtering

* add backend print for g-api launcher

dkustikx committed:
models-dir run_tests.py argument (#2818)

* models-dir argument

* models-dir argument

eaidova committed:
AC configs: fix typo for references (#2829)

Commit:
update hashes

Commit:
update CONTRIBUTING.md

Commit:
Fixed unregistered converter for facial-landmarks-98-detection-0001 (#2831)

Commit:
Refactoring custom evaluators (#2827)

Commit:
Update list of supported models for object detection C++ demo (#2836)

* update list for ssd detection c++ demo models and micro change in async pipeline

* update test cases

* update model lst

* define default meta in async pipeline submit data

Commit:
Fixed SuperResolutionFeedbackEvaluator (#2838)

anzhella-pankratova committed:
Update demos gifs with new statistic, reduce gifs size  (#2833)

* Update gifs, add new example gifs/pictures

* Updates

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -34,64 +34,76 @@ class BatchProducer {
+
+    float getGap(std::chrono::steady_clock::time_point& time_point) {
+        /** Getting of current time **/
+        const auto cur_time = std::chrono::steady_clock::now();
```
`time_point` is not usable here

mpashchenkov replied:
Yes. One more commit is missed. Thanks.

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -207,12 +199,14 @@ int main(int argc, char *argv[]) {
 
             /** Share id with graph **/
             if (size_t(id) < out_detections.size()) {
```
why doesn't make `id` as `size_t` initially? to avoid such casts

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -207,12 +199,14 @@ int main(int argc, char *argv[]) {
             /** Share id with graph **/
             if (size_t(id) < out_detections.size()) {
-                *current_person_id = id;
+                current_person_id1 = id;
```
OPTIONAL: i suggest to move `id` processing into appropriate `else if` to keep it in one place
or, please, check comment below for `wrong number protection`

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp:
```diff
@@ -146,8 +144,8 @@ GAPI_OCV_KERNEL_ST(OCVConstructClip, custom::ConstructClip, BatchState::Params)
-            if (size_t(*current_person_id) < tracked_persons.size()) { // wrong number protection
-                state.last_id = *current_person_id;
+            if (size_t(current_person_id1) < tracked_persons.size()) { // wrong number protection
+                state.last_id = current_person_id1;
```
OPTIONAL:
i see that protection is not strong here:
when you read `atomic` value inside condition load you will get value which atomic have got at the moment of reading
but when you set it to `state.last_id` you will read it again and in general case you can get another value (which may be `wrong`)
So strong protection should be done in a way

```
current_id = current_person_id1.load();
if (current_id < ....) {
    state.last_id = current_id ;
}
```

In latter case double checking (in main) & protection ( here) looks superfluous

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/main.cpp:
```diff
@@ -173,18 +162,21 @@ int main(int argc, char *argv[]) {
             /** Put FPS to frame**/
-            putTimingInfoOnFrame(out_frame, overallTimeAverager.getAveragedValue());
+            if (isStart) {
+                metrics.update(startTime, out_frame, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX,
```
OPTIONAL:
is could have been
`metrics.update(isStart ? startTime : {}, ...)`

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -34,64 +34,76 @@ class BatchProducer {
+            fillFastFrame(ptrI);
+            one_is_ready = true;
+        }
 
```
OPTIONAL:

it could have been take the form:

```
bool one_is_ready = false;
one_is_ready |= try_fillFastFrame(ptrI, batch_read_time);
one_is_ready |= try_fillBatch(ptrI, batch_read_time);

...

try_fill*** () {
    /* division 1000*/
}

getGap() - removed

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -34,64 +34,76 @@ class BatchProducer {
-        fillFastFrame(ptrI);
+        bool one_is_ready = false;
+        /** Counting of gap between last adding of frame to fast frame memory **/
+        const auto frame_gap = getGap(batch_read_time);
```
maybe `frame_read_time` ?

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -34,64 +34,76 @@ class BatchProducer {
+            /** case when batch isn't filled **/
+            return images_in_batch_count++ * frame_size.area() * 3;
         } else {
+            /** Cyclic buffer if filled. Counting of step for next image **/
```
If you use term of `cyclic buffer` then it is more natural to give the following variable names instead current one:

`images_in_batch_count` -> `tail_image_index`
`first_el` -> `head_image_index`

But may be I wrong.

if not, then the more natural  circular buffer processing looks like to do not compare `first_el` with `batch_size -1` but to compare `first_el > images_in_batch_count`

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -34,64 +34,76 @@ class BatchProducer {
+
+    void updateTime(std::chrono::steady_clock::time_point& time_point) {
+        /** Time is updated **/
+        time_point = std::chrono::steady_clock::now();
```
OPTIONAL:
one function for one piece of row of code looks redundant

sivanov-work replied:
in this case renaming `frame_read_time`/`batch_read_time` to `last_frame_timestamp` looks more self-documented

sivanov-work commented on demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp:
```diff
@@ -34,64 +34,76 @@ class BatchProducer {
     bool fillStreamInput() {
-        /** Reading of frame with ImagesCapture class if isn't first_pulled **/
+        /** Reading of frame with ImagesCapture class **/
         cv::Mat frame = cap->read();
```
OPTIONAL:
From my perspective it is more convenient to brake down `BatchProducer` and `CustomCapSource ` dependency according to GRASP principles `Low coupling` by removing aggregating of CustomCapSource  by BatchProducer as member and reuse it in the way:

```
CustomCapSource  {

pull () {
...
cv::Mat frame = cap.read();
 if (!producer.fillStreamInput(frame)) {
            return false;
 }
};
```
where
```
BatchProducer {

fillStreamInput(cv::Mat frame)
/* 
No cap-related logic
only frame & batch processing
*/
}
```
Till dependency between classes is broken down  we will reach idea of reusable stand-alone `batch-container` class (or `frame-throttler` in strong terms)


Please feel free to ignore this comment as optional

sivanov-work reviewed (commented):
Added mostly `optional` comments

Commit:
Added path extending via pkgutil for openvino/tools in accuracy_checker (#2843)

Commit:
add tid (#2842)

Commit:
AC: Fixed adapter type setting in custom evaluators (#2841)

* Fixed setting adapter type in custom evaluators

* Fixed misprints

Wovchena committed:
yaml.load -> yaml.safe_load

Commit:
Added yolo-v3 and yolo-v3-tiny ONNX models (#2755)

Wovchena committed:
Merge pull request #2846 from Wovchena/yaml.load-yaml.safe_load

yaml.load -> yaml.safe_load

Wovchena committed:
quartznet-15x5-en/model.py: use safe load through YAML = YAML(typ='safe') (#2848)

vladimir-dudnik committed:
yolo-v4-tiny-tf now work on MyriadX (#2849)

Wovchena committed:
yaml.load -> yaml.safe_load

Wovchena committed:
Merge pull request #2850 from Wovchena/yaml.load-yaml.safe_load

yaml.load -> yaml.safe_load

eaidova committed:
AC: fix threshold for coco metrics (#2851)

Commit:
Fixed data layout for COCO dataset (#2839)

eaidova committed:
debug pickle loading (#2852)

* debug pickle loading

* change ;pot dependencies order

Commit:
rx check

eaidova committed:
AC: fix coco orig metrics (#2856)

eaidova committed:
AC: fix number fields for Amazon dataset (#2857)

eaidova committed:
AC: prevent devision by zero in wer metrics (#2858)

eaidova committed:
AC: allow disabling comparing with ref (#2859)

Commit:
Add FBCNN jpeg artifact removal model (#2814)

Commit:
Add scipy to requirements of modelAPI (#2854)

eaidova committed:
AC: fix segment_ids for tokenized text (#2861)

eaidova committed:
AC fix padding for squad dataset (#2865)

eaidova committed:
AC: update instance segmentation profiling (#2862)

eaidova committed:
Demo docs: update link on openvino docs (#2860)

Commit:
hashes for new models

Commit:
update schema

eaidova committed:
AC: fix usage glob for kaldi feats (#2867)

Commit:
add requirements

Commit:
update setup.cfg

eaidova committed:
AC: support labels shift for greedy search decoding (#2869)

Wovchena committed:
models/public/drn-d-38/model.yml: update google_drive id

Wovchena committed:
demos/cpp: align gflags with openvino (#2855)

eaidova committed:
AC: update BLEU metric support persample calculation (#2871)

* AC: update BLEU metric support persample calculation

* fix wrong indentation

Wovchena committed:
demos/cpp/detection_model_yolo: use ngraph's alias

Wovchena committed:
Merge pull request #2873 from Wovchena/demos/cpp/detection_model_yolo-ngraph-ov

demos/cpp/detection_model_yolo: use ngraph's alias

eaidova committed:
AC: support word embeddings similarity (#2874)

Wovchena committed:
demos: enable a custom build output folder

https://github.com/openvinotoolkit/openvino/pull/8263

eaidova committed:
AC: fix instance segmentation profiling for not matched (#2879)

Wovchena committed:
Merge pull request #2877 from Wovchena/build_scripts

demos: enable a custom build output folder

Wovchena committed:
Merge remote-tracking branch 'upstream/develop' into models/public/drn-d-38/model.yml-update-google_drive-id

Wovchena committed:
Merge pull request #2870 from Wovchena/models/public/drn-d-38/model.yml-update-google_drive-id

models/public/drn-d-38/model.yml: update google_drive id

Wovchena committed:
Merge remote-tracking branch 'upstream/develop' into HEAD

eaidova committed:
AC: update segmentation profiling (#2881)

* AC: update segmentation profiling

* Update tools/accuracy_checker/openvino/tools/accuracy_checker/metrics/semantic_segmentation.py

* Update tools/accuracy_checker/openvino/tools/accuracy_checker/metrics/semantic_segmentation.py

eaidova committed:
AC: support speaker reidentification (#2878)

* AC: support speaker reidentification

* avoid max recursion in subset selection

* add docs and shape calculation

Wovchena committed:
Merge pull request #2830 from jkamelin/jk/sha384

tools/model_tools: Move to sha384

eaidova committed:
AC: fix getting polygons in profiling (#2882)

eaidova committed:
AC: move masks to per class results (#2883)

* AC: move masks to per class results

* prevent empty polygons metrics

Commit:
update to sha384

eaidova committed:
AC: support explicit specifing undefined shapes in config (#2887)

Commit:
Applying comments and aligning with OMZ demos

Commit:
Removed extra file from ifd (mmissed mtriics unificatton part)

Wovchena committed:
Merge pull request #2868 from jkamelin/jk/yaml_validation

model_tools: yaml validation

eaidova committed:
AC: deprocess prediction in profiling mode (#2884)

* AC: deprocess prediction in profiling mode

* fix wrong prediction in profiling

* update postprocessors behaviour

* fix meta for subset

Wovchena committed:
Merge pull request #2778 from mpashchenkov/mp/ifd-ss

 G-API. Common StreamSource. IFD-GAPI small refactoring.

ilya-lavrenov committed:
OMZ demo add check for minimal OpenCV version (#2888)

* OMZ demo add check for minimal OpenCV version

* Updated OpenCV version required for G_API demo

vladimir-dudnik committed:
human_pose_estimation_3d doc (#2840)

* python ext modules should be added to PYTHONPATH

* Update demos/README.md

Co-authored-by: Tatiana Savina <tatiana.savina@intel.com>

* Update demos/human_pose_estimation_3d_demo/python/README.md

Co-authored-by: Tatiana Savina <tatiana.savina@intel.com>

Co-authored-by: Tatiana Savina <tatiana.savina@intel.com>

eaidova committed:
AC: fix quantization evaluator without adapter (#2889)

morkovka1337 committed:
add tr 0016 model

morkovka1337 committed:
fix model name

morkovka1337 committed:
add tr 0016 to index

morkovka1337 committed:
update demo readme

morkovka1337 committed:
add tests

Commit:
update calls of mo and pot executables (#2890)

Commit:
Returned include

Wovchena committed:
Merge pull request #2892 from mpashchenkov/mp/one-more-include

GAPI: Returning include to IFD GAPI.

Commit:
remove comment, add #TODO mark

Commit:
add more todos

Commit:
comment test case

Commit:
G-API: Aligning test cases for g-api demos. (#2896)

* Added morenets

* Unsupported functional

Commit:
change task type for 0016 model

Commit:
change task type for tr 0015

Commit:
Apply suggestions from code review

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

Commit:
upd readme

Commit:
Merge branch 'feature/vl/support_tr_0016' of https://github.com/openvinotoolkit/open_model_zoo into feature/vl/support_tr_0016

Commit:
add details about decoder output

Wovchena committed:
Merge pull request #2897 from openvinotoolkit/release

Merge release into master

Wovchena committed:
Merge pull request #2891 from openvinotoolkit/feature/vl/support_tr_0016

Text Recognition 0016

Wovchena committed:
demos/image_retrieval: update link to gallery.txt

The file was deleted in
https://github.com/openvinotoolkit/training_extensions/pull/660

Wovchena committed:
Merge pull request #2901 from Wovchena/demos/image_retrieval-update-link-to-gallery.txt

demos/image_retrieval: update link to gallery.txt

eaidova committed:
AC: fix initial reshape for dynamic inputs case (#2902)

dkustikx committed:
face-detection-adas-0001 interactive_face_detection_demo (#2903)

anzhella-pankratova committed:
Introduce FrameworkAdapter (#2844)

* Add inference adapter

* Do not refer directly to model_executor attributes

* Move Openvino interface from the demo and AsyncPipeline to Adapter, implement RemoteAdapter, add documentation to ModelAdapter Abstract class

* Add the __call__ method to ModelBase class

* Refactoring

* Update get_input_layers(), get_output_layers(), add Metadata structure, separate the model reading from the loading

* Update model wrappers for object_detection_demo, except YOLO wrappers

* Update Deblurring, Monodepth demos

* Update bert_named_entity_recognition, bert_question_answering demos

* Move logging to OpenvinoAdapter, add class Core

* Ngraph removed from YOLO

* Fix

* Update human_pose_estimation_demo + Openpose wrapper

* Update hpe_associative_embedding wrapper, bert_question_answering_embedding demo

* Fix in object_detection_demo

* Delete extra EOF

* Update yolo.py

* Remove int2float(), add list of openvino precisions

* Small fixes

* Suggestions

* Add ability to read network from byte buffer.

* Update demos/common/python/openvino/model_zoo/model_api/adapters/openvino_adapter.py

* Update demos/common/python/openvino/model_zoo/model_api/adapters/openvino_adapter.py

* Remove trailing whitespaces

Co-authored-by: eizamaliev <eduard.zamaliev@intel.com>

eaidova committed:
AC: support label map from datumaro (#2904)

Wovchena committed:
Merge branch 'develop' into change-branching-strategy

Wovchena committed:
Merge pull request #2905 from Wovchena/change-branching-strategy

Change branching strategy

Align the strategy with https://github.com/openvinotoolkit/openvino/wiki/Branches:
master - development branch
releases/202?/? - release branches

Wovchena committed:
CONTRIBUTING.md: develop->master

Wovchena committed:
Merge pull request #2906 from Wovchena/CONTRIBUTING.md-develop-master

CONTRIBUTING.md: develop->master

Commit:
Fixed max_ind_range configuration in criteo converter (#2908)

Commit:
model_tools: update installation section (#2863)

* update install section

* update openvino-dev installation

* add note about pip version

vladimir-dudnik committed:
correct output format for classification models (#2910)

eaidova committed:
AC: support openvino 2.0 launcher (#2898)

Commit:
Added lost move

Commit:
Changed rights

Commit:
Applying comments

Commit:
Applying comments + added check for negative value of source

Commit:
Removed extra cast

Commit:
Wrapped time_step

Commit:
Applying comments, added dropping batch info, fixed bug, reduced memory usage

Commit:
Warnings

Commit:
Removed mistake

dmatveev commented:
@mpashchenkov can you please close this PR? I believe its no longer relevant?
